### ISSUE TITLE
PI NODE PORT REFUSED OR IT IS NOT DOWNLOADING

### ISSUE THREAD

**Issue by fairowl-om** [pi-node/instructions#343]
I have tried for several times to connect the port to run the blockchain but ports are not listening and it is not even clearing even first checkbox of downloading ports listener, I thinks problem is about my ISP( GTPL) is using NAT and that is the reason why I am not able to ping the specific ports  ( because I have already done the Inbound rule in firewall for 31400-31409 ports with TCP; WSL update, ports forwarding). Is there any way to tackle this issue. I have tried to resolve this from this issue from tech geeks on issue section of pi node section but so far it hasn't worked for me, any help is welcomed on this issue.

**Comment by dearvcc** [comment]
try:
tcping x.x.x.x 31402

**Comment by cheme75** [comment]
Try to manually set port listening, for each port 31400-31409:

netsh interface portproxy add v4tov4 listenaddress=192.168.x.x listenport=31400 connectaddress=192.168.x.y connectport=31400

Notes: 
- x=router/gateway IP 
- y=the docker node pc IP
- Docker 4.12 is older but ran best for me. You have look in prior releases to download older versions. 

If this still doesn't work, try the "user preferences.json" edit so you can bypass the ports checker. Look it up under issues, there's plenty of info on it.

**Comment by fairowl-om** [comment]
@dearvcc  I have tried tcping but ports are showing no response

**Comment by fairowl-om** [comment]
@cheme75 I have also used netsh command for ports but there is no effect of it, I think issue may be around my ISP having enabled NAT which reject the ports activity.

**Comment by cheme75** [comment]
> @cheme75 I have also used netsh command for ports but there is no effect of it, I think issue may be around my ISP having enabled NAT which reject the ports activity.

I’m in US, haven’t run into isp issue. I control my routers internal side and it has always allowed for port forwarding. It’s how I can access my NAS like a personal cloud from outside my network. Possibly try to contact your isp saying your port fwd isn’t working (don’t say for crypto, just say for your external access to personal apps).

**Comment by fairowl-om** [comment]
> > @cheme75 I have also used netsh command for ports but there is no effect of it, I think issue may be around my ISP having enabled NAT which reject the ports activity.
> 
> I’m in US, haven’t run into isp issue. I control my routers internal side and it has always allowed for port forwarding. It’s how I can access my NAS like a personal cloud from outside my network. Possibly try to contact your isp saying your port fwd isn’t working (don’t say for crypto, just say for your external access to personal apps).

I have tried to talk about this with my ISP but they are saying that there is not any single port blocked by ISP. so, I think problem may lies within the router firewall.

**Comment by cheme75** [comment]
> > > @cheme75 I have also used netsh command for ports but there is no effect of it, I think issue may be around my ISP having enabled NAT which reject the ports activity.
> 
> > 
> 
> > I’m in US, haven’t run into isp issue. I control my routers internal side and it has always allowed for port forwarding. It’s how I can access my NAS like a personal cloud from outside my network. Possibly try to contact your isp saying your port fwd isn’t working (don’t say for crypto, just say for your external access to personal apps).
> 
> 
> 
> I have tried to talk about this with my ISP but they are saying that there is not any single port blocked by ISP. so, I think problem may lies within the router firewall.
> 
> 

Just seems odd. Port forwarding has worked fine whether it was my own router or the isp router. My last 3 over the last 5-6 yrs have been isp provided and none have blocked any port forwarding. I'm out of ideas.

**Comment by Chippleadder** [comment]
are you talking about like a t-mobile home internet connection? if it is or verizon even at&t has one now it has a 5g connection to their mobile network and uses what is called Carrier Grade NAT or CGNAT. With this type of connection they pool IP addresses with a lot of other users. So they block all port forwarding at the modem coming in for you do not have a static ip address so they cannot route directly to you. I had this problem. There is only one way that I found around it and it cost a little bit of money to setup but not much. You must get a VPS server for about 5 bucks a month and then set up a wireguard VPN tunnel (free) and then ssh into the remote machine and set this up. I can provide a link to a tutorial that is in German that walks you through the whole process once you translate it. But warning not for the nontech savy and is difficult even with some coding or DOS/cmd prompt experience. Not certain if ok to post link here maybe someone here knows if it is ok??? But if you are up for it the results are all ports forwarded correctly on my machine and I am on tmobile home internet 5g with CGNAT and ports blocked by isp. I am however still having issues getting blockchain to sync as it seems several others are as well. hope this helps...
